### PR TITLE
Deprecate RecorderContext#classProxy method

### PR DESCRIPTION
**Pull request by jaikiran**
As discussed in the Quarkus dev mailing list, the `RecorderContext#classProxy` is no longer needed since the build steps can now safely load the application/deployment classes without the need for this proxy construct.

This PR has 3 commits - one which does the deprecation and a couple of others which remove the usage of `classProxy` from qute and smallrye-health extensions. The updates to qute and smallrye-health are here to make sure things do actually work fine if the usage of `classProxy` method is replaced. If this shows no regressions, then as separate PRs we can move some of the remaining extensions in this repo, away from this method.

### CODE REVIEW

File: extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/MessageBundleProcessor.java [hunk 1]
```diff
@@ -236,11 +237,17 @@ AdditionalBeanBuildItem beans() {
...
 
         Map<String, Map<String, Class<?>>> bundleInterfaces = new HashMap<>();
         for (MessageBundleBuildItem bundle : bundles) {
-            Class<?> bundleClass = context.classProxy(bundle.getDefaultBundleInterface().toString());
+            final Class<?> bundleClass;
+            try {
+                bundleClass = Class.forName(bundle.getDefaultBundleInterface().toString(), true,
+                        Thread.currentThread().getContextClassLoader());
+            } catch (ClassNotFoundException e) {
+                throw new BuildException(e, Collections.emptyList());
```
Comment (stuartwdouglas) [inline extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/MessageBundleProcessor.java:249]: I would just throw the CNFE, I don't think wrapping it adds anything

Comment (jaikiran) [reply extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/MessageBundleProcessor.java:249]: Sounds fine. I'll update it accordingly.

Comment (jaikiran) [reply extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/MessageBundleProcessor.java]: Done. PR updated.

**Comment by jaikiran** [comment]
The failures so far are unrelated to this change. The one in `ConcurrentExecutionTest` is being tracked here https://github.com/quarkusio/quarkus/pull/10444.
The other failures are in `KafkaCodecTest` which is failing with:
```
2020-07-03T03:35:22.1095830Z Caused by: java.lang.RuntimeException: java.lang.IllegalArgumentException: Could not find a 'KafkaServer' or 'client.KafkaServer' entry in the JAAS configuration. System property 'java.security.auth.login.config' is /home/runner/work/quarkus/quarkus/integration-tests/kafka/target/kafka-server-jaas.18011188515337623419.conf
2020-07-03T03:35:22.1096049Z 	at io.quarkus.it.kafka.KafkaSASLTestResource.start(KafkaSASLTestResource.java:46)
2020-07-03T03:35:22.1096219Z 	at io.quarkus.test.common.TestResourceManager.start(TestResourceManager.java:52)
2020-07-03T03:35:22.1096368Z 	... 45 more
2020-07-03T03:35:22.1541184Z Caused by: java.lang.IllegalArgumentException: Could not find a 'KafkaServer' or 'client.KafkaServer' entry in the JAAS configuration. System property 'java.security.auth.login.config' is /home/runner/work/quarkus/quarkus/integration-tests/kafka/target/kafka-server-jaas.18011188515337623419.conf
2020-07-03T03:35:22.1541844Z 	at org.apache.kafka.common.security.JaasContext.defaultContext(JaasContext.java:133)
2020-07-03T03:35:22.1542375Z 	at org.apache.kafka.common.security.JaasContext.load(JaasContext.java:98)
2020-07-03T03:35:22.1542594Z 	at org.apache.kafka.common.security.JaasContext.loadServerContext(JaasContext.java:70)
2020-07-03T03:35:22.1542773Z 	at 
```
Not yet sure what changed in that area recently.

**Review by Sanne — APPROVED** [review]
Great, thanks @jaikiran !

**Comment by jaikiran** [comment]
> The other failures are in KafkaCodecTest 

I created https://github.com/quarkusio/quarkus/issues/10469 to track this.

**Pull request merged**